linux/LinuxProcess.c

```diff
@@ -189,6 +198,279 @@ void LinuxProcess_printDelay(float delay_percent, char* buffer, int n) {
+(not just the comm prefix) is colorized for better readability, and it is
+implicit that only upto (TASK_COMM_LEN - 1) could be comm
+*/
+static inline bool findCommInCmdline(const char *comm, char *cmdline, int cmdlineBasenameOffset, int *pCommStart, int *pCommEnd) {
+   /* Try to find procComm in tokenized cmdline - this might in rare cases
+    * mis-identify a string or fail, if comm or cmdline had been unsuitably
+    * modified by the process */
+   bool lastToken = false;
+   char *token, *tokenBase;
+
+   for (token = cmdline + cmdlineBasenameOffset; *token; ) {
+      for (tokenBase = token; *token && *token != '\n'; ++token) {
+         if (*token == '/')
+            tokenBase = token + 1;
+      }
+      if (*token)    /* NUL terminate at delimiter for strncmp */
+         *token = 0;
+      else
+         lastToken = true;
+
+      if (strncmp(tokenBase, comm, TASK_COMM_LEN - 1) == 0) {
+         if (!lastToken)   /* restore the delimiter */
+            *token = '\n';
+         *pCommStart = tokenBase - cmdline;
+         *pCommEnd = token - cmdline;
+         return true;
+      }
+      if (!lastToken) {
+         /* restore the delimiter and skip repeats */
+         *token = '\n';
+         while (*++token == '\n')
+            ;
+      }
+   }
+   return false;
+}
+
+static inline int matchCmdlinePrefixWithExeSuffix(const char *cmdline, int cmdlineBaseOffset, const char *exe, int exeBaseOffset, int exeBaseLen) {
+   int matchLen;       /* matching length to be returned */
+   char delim;         /* delimiter following basename */
+
+   /* cmdline prefix is an absolute path: it must match whole exe. */
+   if (cmdline[0] == '/') {
+      matchLen = exeBaseLen + exeBaseOffset;
+      if (strncmp(cmdline, exe, matchLen) == 0 &&
+          ((delim = cmdline[matchLen]) == 0 || delim == '\n' || delim == ' '))
+         return matchLen;
+
+      return 0;
+   }
+
+   /* cmdline prefix is a relative path: We need to first match the basename at
+    * cmdlineBaseOffset and then reverse match the cmdline prefix with the exe
+    * suffix. But there is a catch: Some processes modify their cmdline in ways
+    * that make htop's identification of the basename in cmdline unreliable.
+    * For e.g. /usr/libexec/gdm-session-worker modifies its cmdline to
+    * "gdm-session-worker [pam/gdm-autologin]" and htop ends up with
+    * procCmdlineBasenameOffset at "gdm-autologin]". This issue could arise with
+    * chrome as well as it stores in cmdline its concatenated argument vector,
+    * without NUL delimiter between the arguments (which may contain a '/')
+    *
+    * So if needed, we adjust cmdlineBaseOffset to the previous (if any)
+    * component of the cmdline relative path, and retry the procedure. */
+   bool delimFound;  /* if valid basename delimiter found */
+   do {
+      /* match basename */
+      matchLen = exeBaseLen + cmdlineBaseOffset;
+      if (cmdlineBaseOffset < exeBaseOffset &&
```

Comment:
Could `<=` work here too for the offset comparison?

Reply:
No, <= not needed, since cmdline[0] == '/' was already taken care off, cmdlineBaseOffset must be strictly < exeBaseOffset (as exe is guaranteed to start with '/').

Reply:
k, was just unsure about the rationale.